pkg/util/util.go

```diff
@@ -210,3 +215,20 @@ func MapCopy(oldMap map[string]string) map[string]string {
 	}
 	return newMap
 }
+
+// IdentifyImproperlyFormattedParameter is called if the JSON marshal fails to parse the parameter map.
+// It attempts to convert each individual parameter into a JSON string to identify the specific
+// parameter that is improperly formatted on the storage class. This allows us to provide a more
+// granular error message to the user.
+func identifyImproperlyFormattedParameter(parameters map[string]any) error {
```

Comment:
Can you add some unit tests for this function? Other than that this looks good to me

Reply:
Added!